internal/service/network/network.go

```diff
@@ -4,291 +4,395 @@ import (
+	"github.com/hashicorp/terraform-plugin-framework-validators/listvalidator"
+	"github.com/hashicorp/terraform-plugin-framework-validators/setvalidator"
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/boolplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/setplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
 
-	"github.com/hashicorp/go-cty/cty"
+var (
+	_ resource.Resource                = &networkResource{}
+	_ resource.ResourceWithConfigure   = &networkResource{}
+	_ resource.ResourceWithImportState = &networkResource{}
 )
 
-func ResourceNetwork() *schema.Resource {
-	return &schema.Resource{
-		Description:   "This resource represents an SDN private network that cloud servers from the same zone can be attached to.",
-		ReadContext:   resourceNetworkRead,
-		CreateContext: resourceNetworkCreate,
-		UpdateContext: resourceNetworkUpdate,
-		DeleteContext: resourceNetworkDelete,
-		Importer: &schema.ResourceImporter{
-			StateContext: schema.ImportStatePassthroughContext,
-		},
-		Schema: map[string]*schema.Schema{
-			"ip_network": {
-				Type:        schema.TypeList,
+func NewNetworkResource() resource.Resource {
+	return &networkResource{}
+}
+
+type networkResource struct {
+	client *service.Service
+}
+
+func (r *networkResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_network"
+}
+
+// Configure adds the provider configured client to the resource.
+func (r *networkResource) Configure(_ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*service.Service)
+
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected resource Configure type",
+			fmt.Sprintf("Expected *service.Service, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	r.client = client
+}
+
+type networkModel struct {
+	Name      types.String     `tfsdk:"name"`
+	ID        types.String     `tfsdk:"id"`
+	Type      types.String     `tfsdk:"type"`
+	Zone      types.String     `tfsdk:"zone"`
+	Router    types.String     `tfsdk:"router"`
+	IPNetwork []ipNetworkModel `tfsdk:"ip_network"`
```

Comment:
Try to use `types.Object` here to avoid failing validation when using `some_list[count.index]` to set `Address` value, see [discussion](https://discuss.hashicorp.com/t/how-count-index-works/59139).

Reply:
Done ✅